application/src/main/kotlin/net/corda/v5/application/marshalling/JsonWriter.kt

```diff
@@ -0,0 +1,482 @@
+package net.corda.v5.application.marshalling
+
+import java.io.InputStream
+import java.math.BigDecimal
+import java.math.BigInteger
+
+/**
+ * An interface to a creator of Json. Part of the support to add custom serializer support to the
+ * [JsonMarshallingService]. Custom serializers are passed a [JsonWriter] in order they can translate their object
+ * types to Json.
+ *
+ * When writing arrays and objects, methods which denote the start and end are provided, which should enclose calls to
+ * write any content within those arrays or objects. The root of the Json being serialized should always start with a
+ * call to [writeStartObject] and end with a call to [writeEndObject]. Arrays of course can only contain values, not
+ * fields. Objects must contain fields and values. Separators between objects, arrays, and array items are automatically
+ * added.
+ *
+ * When fields and values are written with either two method calls or the single methods which take a field name and
+ * value parameter, the separators and enclosing double quotation marks are added automatically. The [JsonWriter] will
+ * throw if you attempt to write values without field names, the exception being the [writeRaw] methods which dump
+ * whatever you pass directly into the Json output unchanged. To write raw content with validation instead call one of
+ * the [writeRawValue] methods, which will assume whatever is passed is a valid Json value.
+ */
+interface JsonWriter {
```

Comment:
Might be worth sticking a `@DoNotImplement` on here to signal that we give you one of these and you're not expected to provide one yourself.